core/example/tutorial/01_hello_world/hello_world.cpp

```diff
@@ -18,35 +18,15 @@
 //---------------------------------------------------------------------------//
 int main( int argc, char* argv[] )
 {
-    /* The Cabana runtime must be initialized. This is ultimately to ensure
-       the initialization of the Kokkos runtime that Cabana uses as an
-       implementation detail. Some notes:
-
-           *: If Kokkos has already been initialized by some other part of the
-              application then this initialization does nothing
-
-           *: If Kokkos has not yet been initialized then Cabana will do the
-              initialization.
-
-       You have two options for initialization:
-
-           1) Call Cabana::initialize() with argc and argv as below. This will
-           pass any command line argument to the Kokkos runtime.
-
-           2) Call Cabana::initialize() with no arguments. Defaults and system
-           variables will be used to select things like number of threads.
-    */
-    Cabana::initialize(argc,argv);
+    // The Kokkos runtime must first be initialized.
+    Kokkos::initialize(argc,argv);
```

Comment:
Yeah let's use the `Kokkos::ScopeGuard` here

Reply:
Macro and ScopeGuard added

Reply:
@streeve you misunderstood how the scope guard works.  It goes like this
```C++
namespace Kokkos {
struct ScopeGuard {
  ScopeGuard(int &argc, char * argv[]) { initialize(argc, argv); }
  ~ScopeGuard() { finalize(); }
};
} // namespace Kokkos
```
So you would actually do:
```diff
- Kokkos::initialize(argc, argv);
+ Kokkos::ScopeGuard scope_guard(argc, argv);
  // do stuff
- Kokkos::finalize();
```
Kokkos is initialized when `scope_guard` is constructed and when it goes out of scope its destructor is called which guarantees that Kokkos gets finalized.
In practice `Kokkos::ScopeGuard` implementation is a bit smarter that what I showed you but you get the gist of it.